service/storage_service.cc

```diff
@@ -598,6 +598,7 @@ class topology_coordinator {
         const replica_state* rs;
         std::optional<topology_request> request;
         std::optional<request_param> req_param;
+        raft::server_id replaced_node;
```

Comment:
You mean include a field for each possible param type (`sstring rebuild_option` and so on)? That would be overkill I think.

I included `replaced_node` here since it's needed by `exec_global_command`, and I'm moving `get_node_to_work_on` calls inside `switch` cases in `transition` (and in the CDC PR, some of the cases won't call `get_node_to_work_on`), I didn't want to repeat another additional line for parsing `replaced_node` in each case which has a `get_node_to_work_on` call.

However maybe a better way would be to do the parsing for `replaced_node` inside `exec_global_command`?

Reply:
> 
> However maybe a better way would be to do the parsing for `replaced_node` inside `exec_global_command`?

This is how it is done now, no?